ouroboros-consensus-cardano/app/DBAnalyser/Parsers.hs

```diff
@@ -159,6 +160,18 @@ pMaybeOutputFile =
       <> completer (bashCompleter "file")
       )
 
+parseMaybe ::  Parser a -> Parser (Maybe a)
+parseMaybe parser = asum [Just <$> parser, pure Nothing]
```

Comment:
Minor: As you are touching this: This could be deleted and replaced by [`optional`](https://hackage.haskell.org/package/base-4.19.0.0/docs/Control-Applicative.html#v:optional) at call sites.